arelle/plugin/validate/NL/rules/nl_kvk.py

```diff
@@ -1322,6 +1325,54 @@ def rule_nl_kvk_4_4_2_3(
         )
 
 
+@validation(
+    hook=ValidationHook.XBRL_FINALLY,
+    disclosureSystems=[
+        DISCLOSURE_SYSTEM_NL_INLINE_2024
+    ],
+)
+def rule_nl_kvk_4_4_3_1(
+        pluginData: PluginValidationDataExtension,
+        val: ValidateXbrl,
+        *args: Any,
+        **kwargs: Any,
+) -> Iterable[Validation]:
+    """
+    NL-KVK.4.4.3.1: The extension taxonomy MUST not modify (prohibit and/or override) default members assigned to dimensions by the KVK taxonomy
+    """
+    for modelLink in cast(list[ModelLink], val.modelXbrl.baseSets[XbrlConst.dimensionDefault, None, None, None]):
+        if not pluginData.isExtensionUri(modelLink.modelDocument.uri, val.modelXbrl):
+            continue
+        for linkChild in modelLink:
+            if (
+                    isinstance(linkChild,(ModelObject,PrototypeObject))
+                    and linkChild.get(XbrlConst.qnXlinkType.clarkNotation) == "arc"
+                    and linkChild.get(XbrlConst.qnXlinkArcRole.clarkNotation) == XbrlConst.dimensionDefault
+            ):
+                fromLabel = linkChild.get(XbrlConst.qnXlinkFrom.clarkNotation)
+                for fromResource in modelLink.labeledResources[fromLabel]:
+                    if not pluginData.isExtensionUri(fromResource.modelDocument.uri, val.modelXbrl):
+                        yield Validation.error(
+                            codes='NL.NL-KVK.4.4.3.1.extensionTaxonomyOverridesDefaultMembers',
+                             msg=_('A default member does not match the default member settings of the taxonomy. '
+                                   'Update the default member to taxonomy defaults.'
+                                   ),
+                            modelObject=val.modelXbrl.modelDocument
+                        )
```

Comment:
If we're only going to pass the document as the ref, then I think we should ensure that we only fire this once per document. Alternatively, we could fire it per occurrence like we are here, but in that case we should pass the offending element.

Below as well.

Reply:
For consistency with the message I change the modelObject to the element.